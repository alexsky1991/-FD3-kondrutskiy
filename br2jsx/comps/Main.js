import React from 'react';

export default class Main extends React.Component {

	constructor(props) {
	    super(props);

	    const text = `<div>${this.props.text}</div>`;
	    const doc = new DOMParser().parseFromString(text, "application/xml");

	    const nodes = doc.childNodes[0].childNodes;

	    this.result = Object.values(nodes).map((value, i) => {

	      if(value.localName == 'br')
	     	return <br key={i}/>
			
	      return value.data
	    });

	}


  render() {
    return <div className="br2jsx">{this.result}</div>
  }

}

