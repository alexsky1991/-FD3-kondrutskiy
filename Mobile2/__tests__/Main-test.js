"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import Main from '../comps/main';

test('работа Main', () => {

  const component = renderer.create(
    <Main />
  );


  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

   const buttonElem = component.root.find( el => el.id=='pres' ); 
 
  buttonElem.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();


});
 