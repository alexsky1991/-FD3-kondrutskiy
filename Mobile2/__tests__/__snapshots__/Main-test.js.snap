// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Main 1`] = `
<div
  className="wrapper"
>
  <div
    id="pres"
  >
    pres
  </div>
  <div>
    <div
      className="table_panel"
      onClick={[Function]}
    >
      <div
        className="table_panel_button"
        id="all"
      >
        Все
      </div>
      <div
        className="table_panel_button"
        id="active"
      >
        Активные
      </div>
      <div
        className="table_panel_button"
        id="blocked"
      >
        Заблокированные
      </div>
    </div>
    <div
      className="table"
    >
      <div
        className="table_line line_title"
      >
        <div
          className="table_item"
        >
          Фамилия
        </div>
        <div
          className="table_item"
        >
          Имя
        </div>
        <div
          className="table_item"
        >
          Отчество
        </div>
        <div
          className="table_item"
        >
          Баланс
        </div>
        <div
          className="table_item"
        >
          Статус
        </div>
        <div
          className="table_item"
        >
          Редактировать
        </div>
        <div
          className="table_item"
        >
          Удалить
        </div>
      </div>
      <div
        className="table_line"
      >
        <div>
          Иванов
        </div>
        <div>
          Иван
        </div>
        <div>
          Иванович
        </div>
        <div>
          200
        </div>
        <div
          className=" active"
        >
           active
        </div>
        <div>
          <div
            className="table_item_button"
            onClick={[Function]}
          >
            Редактировать
          </div>
        </div>
        <div>
          <div
            className="table_item_button"
            id="add"
            onClick={[Function]}
          >
            Удалить
          </div>
        </div>
      </div>
      <div
        className="table_line"
      >
        <div>
          Сидоров
        </div>
        <div>
          Сидор
        </div>
        <div>
          Сидорович
        </div>
        <div>
          250
        </div>
        <div
          className=" active"
        >
           active
        </div>
        <div>
          <div
            className="table_item_button"
            onClick={[Function]}
          >
            Редактировать
          </div>
        </div>
        <div>
          <div
            className="table_item_button"
            id="add"
            onClick={[Function]}
          >
            Удалить
          </div>
        </div>
      </div>
      <div
        className="table_line"
      >
        <div>
          Петров
        </div>
        <div>
          Петр
        </div>
        <div>
          Петрович
        </div>
        <div>
          180
        </div>
        <div
          className=" active"
        >
           active
        </div>
        <div>
          <div
            className="table_item_button"
            onClick={[Function]}
          >
            Редактировать
          </div>
        </div>
        <div>
          <div
            className="table_item_button"
            id="add"
            onClick={[Function]}
          >
            Удалить
          </div>
        </div>
      </div>
      <div
        className="table_line"
      >
        <div>
          Григорьев
        </div>
        <div>
          Григорий
        </div>
        <div>
          Григорьевич
        </div>
        <div>
          -200
        </div>
        <div
          className=" blocked"
        >
           blocked
        </div>
        <div>
          <div
            className="table_item_button"
            onClick={[Function]}
          >
            Редактировать
          </div>
        </div>
        <div>
          <div
            className="table_item_button"
            id="add"
            onClick={[Function]}
          >
            Удалить
          </div>
        </div>
      </div>
    </div>
    <div
      className="table_add"
      onClick={[Function]}
    >
      Добавить клиента
    </div>
  </div>
</div>
`;
